nixos/modules/services/hardware/amdgpu-fan.nix

```diff
@@ -0,0 +1,73 @@
+{ config, lib, pkgs, ... }:
+with lib;
+
+let
+  cfg = config.services.amdgpu-fan;
+  pkg = pkgs.amdgpu-fan;
+  amdgpuFanConfigFile = pkgs.writeText "amdgpu-fan.yml" ''
+    # speed matrix
+    # -[temp(*C), speed(0-100%)]
+    ${cfg.speed_matrix}
+
+    # optional list of cards
+    ${cfg.cards}
+  '';
+
+in
+{
+
+  options.services.amdgpu-fan = {
+    enable = mkEnableOption "Python daemon for controlling the fans on amdgpu cards";
+
+    speed_matrix = mkOption {
+      type = types.lines;
```

Comment:
A more appropriate type would be better, e.g. `uniq (listOf (listOf float))` (or are only `int`s possible?). Then you can set these with
```nix
{
  speedMatrix = [ [0 0] [40 30] [60 50] [80 100] ];
}
```

You'll have to write a small generator function for converting this to strings though.